### ISSUE TITLE
problems with make failing, any clues?

### ISSUE THREAD

**Issue by majik12345** [varnish/libvmod-geoip#3]
make  all-recursive
make[1]: Entering directory `/home/libvmod-geoip'
Making all in src
make[2]: Entering directory`/home/libvmod-geoip/src'
/bin/sh ../libtool --tag=CC   --mode=compile gcc -std=gnu99 -DHAVE_CONFIG_H -I. -I..  -I/usr/include/varnish  -Wall -Wextra -Werror -g -O2 -MT vmod_geoip.lo -MD -MP -MF .deps/vmod_geoip.Tpo -c -o vmod_geoip.lo vmod_geoip.c
libtool: compile:  gcc -std=gnu99 -DHAVE_CONFIG_H -I. -I.. -I/usr/include/varnish -Wall -Wextra -Werror -g -O2 -MT vmod_geoip.lo -MD -MP -MF .deps/vmod_geoip.Tpo -c vmod_geoip.c  -fPIC -DPIC -o .libs/vmod_geoip.o
cc1: warnings being treated as errors
vmod_geoip.c:21: error: ‘enum vcl_event_e’ declared inside parameter list
vmod_geoip.c:21: error: parameter 3 (‘evt’) has incomplete type
vmod_geoip.c: In function ‘vmod_event’:
vmod_geoip.c:27: error: ‘VCL_EVENT_LOAD’ undeclared (first use in this function)
vmod_geoip.c:27: error: (Each undeclared identifier is reported only once
vmod_geoip.c:27: error: for each function it appears in.)
vmod_geoip.c:21: error: unused parameter ‘evt’
make[2]: **\* [vmod_geoip.lo] Error 1
make[2]: Leaving directory `/home/libvmod-geoip/src'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory`/home/libvmod-geoip'
make: **\* [all] Error 2

cant get make to work

here is configure

checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking dependency style of gcc... gcc3
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define **EXTENSIONS**... yes
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking dependency style of gcc... (cached) gcc3
checking for gcc option to accept ISO C99... -std=gnu99
checking for gcc -std=gnu99 option to accept ISO Standard C... (cached) -std=gnu99
checking how to run the C preprocessor... gcc -E
checking for a sed that does not truncate output... /bin/sed
checking for fgrep... /bin/grep -F
checking for ld used by gcc -std=gnu99... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1966080
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for ar... ar
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc -std=gnu99 object... ok
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc -std=gnu99 supports -fno-rtti -fno-exceptions... no
checking for gcc -std=gnu99 option to produce PIC... -fPIC -DPIC
checking if gcc -std=gnu99 PIC flag -fPIC -DPIC works... yes
checking if gcc -std=gnu99 static flag -static works... no
checking if gcc -std=gnu99 supports -c -o file.o... yes
checking if gcc -std=gnu99 supports -c -o file.o... (cached) yes
checking whether the gcc -std=gnu99 linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking whether make sets $(MAKE)... (cached) yes
checking for rst2man... rst2man
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for libGeoIP... yes
checking for ANSI C header files... (cached) yes
checking sys/stdlib.h usability... no
checking sys/stdlib.h presence... no
checking for sys/stdlib.h... no
checking for python3... no
checking for python2.7... no
checking for python2.6... python2.6
checking vsha256.h usability... yes
checking vsha256.h presence... yes
checking for vsha256.h... yes
checking cache/cache.h usability... yes
checking cache/cache.h presence... yes
checking for cache/cache.h... yes
checking for varnishtest... /usr/bin/varnishtest
checking for varnishd... /usr/sbin/varnishd
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating config.h
config.status: config.h is unchanged
config.status: executing depfiles commands
config.status: executing libtool commands

**Comment by majik12345** [comment]
ok looks like maybe its because I am still using varnish 4.0.3 ? I will upgrade to 4.1.3 and try again

**Comment by fgsch** [comment]
That is correct. For 4.0.x you need to use the 4.0 branch.

If you have any problems after upgrading please open a new ticket.

**Comment by majik12345** [comment]
That make fails with 4.0.3, it seems it references something in varnish that is only available from 4.1. But Anyway I upgraded my varnish to 4.1.3 and make worked no problem.

**Comment by fgsch** [comment]
fwiw, I checked travis and everything seems to be OK wrt the 4.0 branch.
Maybe some issue on your side?
